File: src/types/queue.ts
// Queue and review types
export interface QueueItem {
  id: string;
  documentId: string;
  documentTitle: string;
  documentFileType?: string;
  extractId?: string;
  learningItemId?: string;
  question?: string;
  answer?: string;
  clozeText?: string;
  itemType: "document" | "extract" | "learning-item" | "playlist-video";
  priorityRating?: number;
  prioritySlider?: number;
  priority: number;
  dueDate?: string;
  estimatedTime: number; // in minutes
  tags: string[];
  category?: string;
  progress: number; // 0-100
  
  // Playlist interspersion fields
  /** Source identifier, e.g., "playlist:<subscription_id>" */
  source?: string;
  /** Position in queue for interspersion calculation */
  position?: number;
}

export interface ReviewSession {
  id: string;
  startTime: string;
  endTime?: string;
  itemsReviewed: number;
  correctAnswers: number;
  totalTime: number; // in seconds
  streakDays: number;
}

export interface ReviewItem {
  id: string;
  documentId: string;
  documentTitle: string;
  extractId?: string;
  learningItemId?: string;
  itemType: "learning-item";
  priority: number;
  dueDate?: string;
  estimatedTime: number;
  tags: string[];
  category?: string;
  progress: number;
  question: string;
  answer?: string;
  context?: string;
  hint?: string;
  relatedItems?: string[];
}

export enum Rating {
  Again = 1,
  Hard = 2,
  Good = 3,
  Easy = 4,
}

export interface ReviewResult {
  itemId: string;
  rating: Rating;
  timeTaken: number; // in seconds
  newDueDate: string;
  newInterval: number;
  newEaseFactor: number;
}

File: src-tauri/src/models/queue.rs
//! Queue model

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct QueueItem {
    pub id: String,
    pub document_id: String,
    pub document_title: String,
    pub extract_id: Option<String>,
    pub learning_item_id: Option<String>,
    pub question: Option<String>,
    pub answer: Option<String>,
    pub cloze_text: Option<String>,
    pub item_type: String, // "document", "extract", "learning-item"
    pub priority_rating: Option<i32>,
    pub priority_slider: Option<i32>,
    pub priority: f64,
    pub due_date: Option<String>,
    pub estimated_time: i32, // in minutes
    pub tags: Vec<String>,
    pub category: Option<String>,
    pub progress: i32, // 0-100
}
